compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -390,29 +394,32 @@ public void visit(BLangImportPackage importPkgNode) {
         Name orgName;
         Name version;
         PackageID enclPackageID = env.enclPkg.packageID;
-        if (importPkgNode.orgName.value == null || importPkgNode.orgName.value.isEmpty()) {
-            // means it's in 'import <pkg-name>' style
-            orgName = enclPackageID.orgName;
-            version = (Names.DEFAULT_VERSION.equals(enclPackageID.version)) ? new Name("") : enclPackageID.version;
-        } else if (importPkgNode.orgName.value.equals(enclPackageID.orgName.value)) {
-            // means it's in 'import <org-name>/<pkg-name>' style and <org-name> is used to import within same project
+        // The pattern of the import statement is 'import [org-name /] module-name [version sem-ver]'
+        // There cases should be considered here.
+        // 1. import org-name/module-name version
+        // 2. import org-name/module-name
+        //      2a. same project
+        //      2b. different project
+        // 3. import module-name
+        if (!isNullOrEmpty(importPkgNode.orgName.value)) {
             orgName = names.fromIdNode(importPkgNode.orgName);
-            // Here we set the version to enclosing package version. Following cases will be handled properly:
-            // 1) Suppose the import is from the same project, then the enclosing package version set here will be used
-            //    to lookup package symbol cache, avoiding re-defining package symbol.
-            String pkgName = importPkgNode.getPackageName().stream()
-                    .map(id -> id.value)
-                    .collect(Collectors.joining("."));
-            if (this.sourceDirectory.getSourcePackageNames().contains(pkgName)) {
-                version = enclPackageID.version;
+            if (!isNullOrEmpty(importPkgNode.version.value)) {
+                version = names.fromIdNode(importPkgNode.version);
             } else {
-                version = new Name("");
+                String pkgName = importPkgNode.getPackageName().stream()
+                        .map(id -> id.value)
+                        .collect(Collectors.joining("."));
+                if (this.sourceDirectory.getSourcePackageNames().contains(pkgName)) {
+                    version = enclPackageID.version;
+                } else {
+                    version = new Name("");
```

Comment:
is it possible go use the `Names.EMPTY` static value here instead of creating a new instance?

Reply:
Fixed.